doc/docusaurus/docs/delve-deeper/casing-constants.md

```diff
@@ -0,0 +1,125 @@
+---
+sidebar_position: 26
+---
+
+# Case-splitting on constants
+
+:::info
+
+This use of `case` in UPLC was introduced with protocol version 11 and requires
+[Plutus Core version](../essential-concepts/versions#plutus-core-version)
+`1.1.0`.
+
+:::
+
+In UPLC, it is possible to branch on expressions of certain built-in types, like
+`Integer` and `Bool`. This can be done with `case` syntax, which is also used
+for [sums-of-products](./encoding#sums-of-products). Using `case` on built-in
+values may improve script performance and size. 
+The built-in types that `case` currently supports are:
+
+- `bool`
+- `unit`
+- `integer`
+- `list`
+- `pair`
+
+In the future, support for `data` is also planned.
+
+For each type, the allowed branches and their order differs. See [Supported
+Types](#supported-types) for more detail.
+
+## Compiling to `case` in Plinth
+
+When compiling Plinth code with the [option](./plinth-compiler-options)
+`datatypes=BuiltinCasing`, many standard library functions will be compiled into
```

Comment:
We need to mention that this option is going away post hard fork (see the comment on the `DatatypeStyle` data type). Also please make an issue to update this post hard fork.